en/docs/install-and-setup/upgrading-wso2-api-manager/260-to-410/upgrading-from-260-to-410.md

```diff
@@ -0,0 +1,1304 @@
+
+    -   Remove the `migration-resources` directory, which is in the `<API-M_4.1.0_HOME>` directory.
+
+    -   If you ran WSO2 API-M as a Windows Service when doing the identity component migration , then you need to remove the following parameters in the command line arguments section (CMD_LINE_ARGS) of the api-manager.bat file.
+
+        ```
+        -Dmigrate -Dcomponent=identity
+        ```
+
+### Step 4: Migrate the API Manager Components
+
+You have to run the following migration client to update the API Manager artifacts.
+
+1.  Navigate to the [latest release tag (v4.1.0.x)](https://github.com/wso2-extensions/apim-migration-resources/releases) under **Assets** and download the `wso2am-migration-4.1.0.x.zip`. Unzip it to a local directory.
+
+    !!! note
+
+        x of wso2am-migration-4.1.0.x.zip denotes the version number of the most recently-released migration resources.
+
+        The directory where the wso2am-migration-4.1.0.x.zip is unziped will be referred to as `<AM_MIGRATION_CLIENT_HOME>`
+
+2. Copy the `<AM_MIGRATION_CLIENT_HOME>/migration-resources` to the `<API-M_4.1.0_HOME>` directory.
+
+3. Copy the org.wso2.carbon.am.migration-4.1.0.x.jar file in the `<AM_MIGRATION_CLIENT_HOME>/dropins` directory into the `<API-M_4.1.0_HOME>/repository/components/dropins` directory.
+
+4.  Start the API-M server to migrate the API-M components as follows.
+
+    ``` tab="Linux / Mac OS"
+    sh api-manager.sh -Dmigrate -DmigrateFromVersion=2.6.0 -DmigratedVersion=4.1.0
+    ```
+
+    ``` tab="Windows"
+    api-manager.bat -Dmigrate -DmigrateFromVersion=2.6.0 -DmigratedVersion=4.1.0
+    ```
+
+5.  Shutdown the API-M server.
+
+    -   Remove the `org.wso2.carbon.apimgt.migrate.client-4.1.0.x.jar` file, which is in the `<API-M_4.1.0_HOME>/repository/components/dropins` directory.
+
+    -   Remove the `migration-resources` directory, which is in the `<API-M_4.1.0_HOME>` directory.
+
+    !!! note
+
+        Make sure you have copied the tenants userstores if you have configured them in API-M 2.6.0
+
+### Step 5: Re-Index the API Manager artifacts
+
+1. To re-index the API artifacts in the registry, Add the following configuration into the `<API-M_4.1.0_HOME>/repository/conf/deployment.toml` file.
+
+    ```
+    [indexing]
+    re_indexing = 1
+    ```
+
+   Note that you need to increase the value of `re_indexing` by one each time you need to re-index.
+
+
+2. If the `<API-M_4.1.0_HOME>/solr` directory exists, take a backup and thereafter delete it.
+
+3. Restart the server.
+
+    ```tab="Linux / Mac OS"
+        sh api-manager.sh
+    ```
+
+    ```tab="Windows"
+        api-manager.bat
+    ```
+
+!!! important
+    If you use a clustered/distributed API Manager setup, do the above change in deployment.toml of Publisher and Devportal nodes. Make sure to keep a delay between nodes to execute this step to re-index each node, as the database can experience a large load.
```

Comment:
Since we have deprecated publisher and devportal, shouldn't this be changed to control-plane?